utils/common/src/test/java/org/apache/brooklyn/util/text/VersionComparatorTest.java

```diff
@@ -56,40 +35,51 @@ public void testSnapshotSuffixComparison() {
 
     @Test
     public void testComparison() {
```

Comment:
as above, i think the comparators should never return 0 unless the strings are actually equal; someone else has determined equality beforehand, and wants this to give a canonical order on the things it thinks are unequal